### PR TITLE
Fix handling of binary files when using ripgrep scanner

### PR DESCRIPTION
**Pull request by rafeca**
This PR is a follow-up of https://github.com/atom/atom/pull/19348

By pure chance I've found an issue when handling files that have lines with binary content while testing the `ripgrep` scanner on the [`gecko-dev`](https://github.com/mozilla/gecko-dev) repository.

The main problem is found when a file has lines with non-UTF characters it returns a base64 representation of that line in the json output (to not break the JSON formatting). There's more information about it [here](https://docs.rs/grep-printer/0.1.1/grep_printer/struct.JSON.html#text-encoding).

Before this PR, we were not handling this scenario and the parser miserably failed 😅

### CODE REVIEW

File: src/ripgrep-directory-searcher.js [hunk 1]
```diff
@@ -146,8 +150,16 @@ function processSubmatch (submatch, lineText, offsetRow) {
...
+    lineText: cleanResultLine({ text: lineParts.join('\n') })
+  }
+}
+
+function getText (input) {
+  if (input.text) {
+    return input.text
   }
+
+  return Buffer.from(input.bytes, 'base64').toString()
```
Comment (nathansobo) [inline src/ripgrep-directory-searcher.js]: Minor: Putting this in an `else` block would be more readable to me.

File: src/ripgrep-directory-searcher.js [hunk 2]
```diff
@@ -280,7 +292,7 @@ module.exports = class RipgrepDirectorySearcher {
 
           if (message.type === 'begin') {
             pendingEvent = {
-              filePath: message.data.path.text,
+              filePath: getText(message.data.path),
```
Comment (nathansobo) [inline src/ripgrep-directory-searcher.js]: Wow, base64 encoded paths? Awesome.

**Review by nathansobo — APPROVED** [review]
Nice find! ⚡️

**Comment by nathansobo** [comment]
In case anyone is wondering whether there's anything possible in these JSON objects other than `text` and `bytes`, [there isn't](https://github.com/BurntSushi/ripgrep/blob/master/grep-printer/src/jsont.rs#L82).

File: src/ripgrep-directory-searcher.js (hunk 1 above)
Comment (rafeca) [reply src/ripgrep-directory-searcher.js]: I've just used a ternary at the end 😁

**Pull request merged**